native/src/fairseq2n/data/bucket_by_length_data_source.cc

```diff
@@ -151,10 +151,10 @@ bucket_by_length_data_source::reload_position(tape &t, bool strict)
     inner_->reload_position(t, strict);
 }
 
-bool
-bucket_by_length_data_source::is_infinite() const noexcept
+data_source_finitude_type
+bucket_by_length_data_source::get_finitude_type() const noexcept
```

Comment:
Nit: since this is mainly an accessor function, for brevity, you can omit the `get_` prefix and just say `finitude_type()`.